src/string.rs

```diff
@@ -96,6 +93,58 @@ pub fn rewrite_string<'a>(s: &str, fmt: &StringFormat<'a>) -> Option<String> {
     Some(result)
 }
 
+fn backwards_find_whole_non_escaped(graphemes: &Vec<&str>, end: usize) -> usize {
+    // Trivial: \n \r \t \" \'
+    // Hard: \x## \u{######}
+    // Hardest:\\\\\\\\ <- break at the even gap of one of these
+    use std::cmp;
+    // At worst, an escape sequence can begin 9 characters before end:
+    //              \u{ABCDEF}
+    // window_start ^        ^ end
+    let window_start: usize = cmp::max((end as isize)-9, 0) as usize;
+    graphemes[window_start..end]
+        .iter()
+        .rposition(|s| *s == "\\")
+        .map(|eidx| window_start + eidx)
+        .map(|escape_idx| { // Shift the start idx if it has an odd number of \\ escapes
+            if graphemes[escape_idx - 1] == "\\" {
+                match graphemes[..escape_idx]
+                          .iter()
+                          .rposition(|s| *s != "\\")
```

Comment:
Linear in the worst case, look backwards along the whole string until we find a non-\